tests/python/relay/test_op_qnn_batch_matmul.py

```diff
@@ -0,0 +1,245 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import tvm
+import numpy as np
+from tvm import relay
+from tvm.contrib import graph_executor
+from tvm.relay.testing.temp_op_attr import TempOpAttr
+
+# We use llvm target for testing functionality. `llvm` points to an older Intel
+# generation machine, that legalizes to a simple lowering. Therefore, the
+# legalization is overwritten such that it can be skipped and we use the
+# QNNCanonicalizeOps lowering for the testing.
+def legalize_qnn_batch_matmul(attrs, inputs, types):
+    return None
+
+
+def make_requantize_params(input_scale, output_scale, output_zero_point, out_dtype):
+    config = {
+        "input_scale": input_scale,
+        "output_scale": output_scale,
+        "output_zero_point": output_zero_point,
+        "out_dtype": out_dtype,
+    }
+    return config
+
+
+def make_configuration(
+    quantized_x,
+    quantized_y,
+    dtype,
+    x_shape,
+    y_shape,
+    x_zero_point,
+    y_zero_point,
+    x_scale,
+    y_scale,
+    output,
+    out_dtype="int32",
+    requantize=None,
+):
+    config = {
+        "quantized_x": quantized_x,
+        "quantized_y": quantized_y,
+        "dtype": dtype,
+        "x_shape": x_shape,
+        "y_shape": y_shape,
+        "x_zero_point": x_zero_point,
+        "y_zero_point": y_zero_point,
+        "x_scale": x_scale,
+        "y_scale": y_scale,
+        "output": output,
+        "out_dtype": out_dtype,
+        "requantize": requantize,
+    }
+    return config
+
+
+def make_int_configuration(
+    xzero_point_zero=True, yzero_point_zero=True, requantize_output=False, per_channel=False
+):
+    x_shape, y_shape, output_shape = (1, 4, 5), (1, 3, 5), (1, 4, 3)
+    if xzero_point_zero == True:
+        x_zero_point = 0
+    else:
+        x_zero_point = -1
+
+    if yzero_point_zero == True:
+        y_zero_point = 0
+    else:
+        y_zero_point = -1
```

Comment:
Should test on more non-zero zero points.

Reply:
@masahi just curious if changing -1 to other value would be enough or you propose to add other tests. 
Currently there are 5 test cases:
x zp =0, y zp = 0
x zp = -1, y zp = 0
x zp = 0, y zp = -1
x zp =-1, y zp =-1
that covers all flows in `QnnBatchMatmulCanonicalize`

Reply:
I think we should test on larger non-zero zero points, to verify the accuracy.

Reply:
modified zp to 123